network/src/components/AppWrapper.js

```diff
@@ -590,24 +612,45 @@ export const AppWrapper = ({children}) => {
     }
 
 
-  // Default options for vis-network
-  const [dynamicOptions, setDynamicOptions] = useState(options);
-
-  
-
-  // Function to toggle zoom based on Ctrl key press
   const handleKeyDown = (event) => {
-    if (event.key === 'Control' || event.key === 'Meta') {  // Meta is for Cmd on macOS
-      setIsCtrlPressed(true);
+    if (event.key === 'Control' || event.key === 'Meta') { 
+      isCtrlPressedRef.current = true;
+      setIsCtrlPressed(true); 
     }
   };
 
   const handleKeyUp = (event) => {
     if (event.key === 'Control' || event.key === 'Meta') {
+      isCtrlPressedRef.current = false;  // Update ref
       setIsCtrlPressed(false);
     }
   };
 
+  const handleCanvasClick = (event) => {
+    setSelectedNodes([]);  // Deselect all nodes on background click
+  };
+
+  const setupEventListeners = (network) => {
+    network.on('click', (event) => {
+      if (!event.nodes.length) {
+        handleCanvasClick(event);
+      }
+    });
+  };
+  
+  // only grabs the entity nodes (Person/Org)
+  const entitySelection = selectedNodes.filter(nodeId => {
+    const node = nodes.find(n => n.id === nodeId).nodeInfo;
```

Comment:
I get an error on this line when trying to delete 1 or more nodes. 
<img width="862" alt="image" src="https://github.com/user-attachments/assets/3a16465a-c80a-4ae0-9a48-4e9c44972c5f">

Reply:
Should be working now